### DISCUSSION TITLE
Custom User Provider SPI: How to configure scheduler to sync users periodically ?

### DISCUSSION THREAD

**Discussion by vplemyannik** [keycloak/keycloak#15259] [Q&A] [answered]
Hi,

I have created my custom User Provider SPI with import strategy.
So, i implemented ImportSynchronization interface
![Screenshot_3](https://user-images.githubusercontent.com/17518385/199334158-dd2327fe-7a14-4a42-82fc-f98ba31a66f0.png)

There are 2 methods `sync` and `syncSince`

I found out how to call `sync` method - click "Sync all users" button. And it works perfectly.
![Screenshot_2](https://user-images.githubusercontent.com/17518385/199334384-4f648819-81b3-49a3-89ad-fdd34faad6af.png)

But i could not find the way how to configure scheduler that will invoke syncSince method. As i understand scheduler options must be shown automatically if UserProviderFactory implements ImportSynchronization.

From [documentation](https://www.keycloak.org/docs/latest/server_development/#import-implementation-strategy) 
_Once this interface is implemented by the provider factory, the administration console management page for the provider shows additional options. You can manually force a synchronization by clicking a button. This invokes the ImportSynchronization.sync() method. Also, additional configuration options are displayed that allow you to automatically schedule a synchronization. Automatic synchronizations invoke the syncSince() method._

Could someone help me to find out how to find these options or how to make them be shown ?

Keycloak v20.0.0

**Comment by DAHAG-ArisNourbakhsh** [comment, marked as answer]
We also have a custom `UserStorageProviderFactory` that implements `ImportSynchronization` and this seems to be a breaking change/issue with the new keycloak.v2 admin theme. The sync settings are just not there in the UI. 

We are currently still mostly using the legacy `keycloak` theme for the admin UI due to the numerous missing features and bugs so I didn't notice till now but the options vanish if I switch to v2. Try switching to the old theme and see if it's there?

**Reply by vplemyannik** [reply]
Thank for reply!

I've changed theme in realm settings, but it seems to me that nothing had changed 
![image](https://user-images.githubusercontent.com/17518385/200823044-9ff7c00b-ad6e-4ca8-ad67-7377259ca996.png)
Admin UI still looks like theme v2


P.S I have reloaded page and it turned into old theme and options have appeared. @DAHAG-ArisNourbakhsh  Thank you so much !
